pkg/generators/apicast/generator.go

```diff
@@ -1,48 +1,43 @@
 package apicast
 
 import (
+	"fmt"
+
 	saasv1alpha1 "github.com/3scale/saas-operator/api/v1alpha1"
 	"github.com/3scale/saas-operator/pkg/generators"
 	"github.com/3scale/saas-operator/pkg/generators/apicast/config"
-	"github.com/3scale/saas-operator/pkg/generators/common_blocks/grafanadashboard"
-	"github.com/3scale/saas-operator/pkg/generators/common_blocks/hpa"
-	"github.com/3scale/saas-operator/pkg/generators/common_blocks/pdb"
-	"github.com/3scale/saas-operator/pkg/generators/common_blocks/podmonitor"
-	basereconciler "github.com/3scale/saas-operator/pkg/reconcilers/basereconciler/v1"
+	basereconciler "github.com/3scale/saas-operator/pkg/reconcilers/basereconciler/v2"
+	basereconciler_resources "github.com/3scale/saas-operator/pkg/reconcilers/basereconciler/v2/resources"
+	"github.com/3scale/saas-operator/pkg/reconcilers/workloads"
+	"github.com/3scale/saas-operator/pkg/resource_builders/grafanadashboard"
+	"github.com/3scale/saas-operator/pkg/resource_builders/podmonitor"
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
 	"k8s.io/apimachinery/pkg/types"
 )
 
 const (
-	apicastStaging    string = "apicast-staging"
-	apicastProduction string = "apicast-production"
-	apicast           string = "apicast"
+	apicastStaging          string = "apicast-staging"
+	apicastCanaryStaging    string = "apicast-canary-staging"
```

Comment:
The canary resources names should be `canary-X`  like done on Backend?

IE: `canary-apicast-staging`/`canary-apicast-production`

Reply:
they are like in Backend :)
![image](https://user-images.githubusercontent.com/12752291/152166455-743582fe-46e4-4263-9335-102a003b5e55.png)

Reply:
As per decided in slack, we will add `canary` as a suffix to the Deployment name, instead of the current infix. Changed for apicast in 9ea4acd. An upcoming PR will change it for all other workloads.